microservices/object-identifer-strategy.md

```diff
@@ -0,0 +1,140 @@
+
+There is currently no clear requirement for the SDC platform that justifies the additional development and operational complexity that would be required to fully implement URNs. If a requirement emerges later it would be easy to add to the strategy proposed here by prefixing all existing UUIDs with a URN namespace identifier (NID) such as `uk.gov.ons.<type>`
+
+## Worked Example
+The example below illustrate how the use of UUIDs to identify objects would work in practice. Rather than get bogged down in the detail the important thing is to note how the UUIDs are passed around between the three collaborating microservices.
+
+For this example the Collection Instrument service needs to load an offline collection instrument (spreadsheet) into the system so it's associated with the correct reporting unit. The RAS back office UI makes a request to the Survey service to get the list of known surveys so it can present their names to the user:
+
+`GET https://surveysvc/surveys`
+
+```json 
+[{
+    "id": "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87",
+    "shortName": "BRES"
+}]
+```
+
+The user selects the survey applicable to the collection instrument they want to load e.g. BRES. At this point the application has the unique identifier for the BRES survey object and can ask the Collection Exercise service for the list of collection exercises for that survey:
+
+`GET https://collectionexercisesvc/collectionexercises/survey/cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87`
+
+```json
+[{
+    "id": "2c82b85a-2a03-434d-8019-eae05b01c6f4",
+    "name": "201601",
+    "scheduledExecutionDateTime": "2017-05-10 11:55:00Z"
+},
+{
+    "id": "...",
+    "name": "...",
+    "scheduledExecutionDateTime": "..."
+}]
+```
+
+The collection exercise details are displayed to the user, who selects the relevant collection exercise. The Collection Instrument service can now persist the spreadsheet by making an HTTP PUT request to update the relevant collection instrument resource. Note how the request payload contains the UUID for the collection exercise retrieved in the previous step:
+
+`PUT https://collectioninstrumentsvc/collectioninstruments/6A33A6C9-F01F-4224-81D6-362C211DFD31`
+
+```json
+{
+    "classifiers": [{
+        "classifierType": "RU_REF",
+        "classifier": "49900000001"
+    },
+    {
+        "classifierType": "COLLECTION_EXERCISE",
+        "classifier": "2c82b85a-2a03-434d-8019- eae05b01c6f4"
+    }]
+}
+```
```

Comment:
What relevance do the classifiers have in this? I think this worked example may be too specific to how it's been implemented and we should consider creating a more abstract one.

Reply:
That's a fair point. To be honest it's a bit tricky because this strategy was originally written on Confluence and only for the RAS and RM teams because we needed to build this stuff and needed something ASAP. I wanted to use concrete examples that audience could relate to. 

Perhaps we now need to take a step back and think about it more generically for the whole platform. What might an EQ example look like?